File: web/src/plugin/project_manager/api/project.js
import service from '@/utils/request'

export const createProject = (data) => {
  return service({
    url: '/project/createProject',
    method: 'post',
    data
  })
}

export const deleteProject = (data) => {
  return service({
    url: '/project/deleteProject',
    method: 'delete',
    data
  })
}

export const updateProject = (data) => {
  return service({
    url: '/project/updateProject',
    method: 'put',
    data
  })
}

export const getProjectList = (params) => {
  return service({
    url: '/project/getProjectList',
    method: 'get',
    params
  })
}

File: server/plugin/project_manager/service/project.go
package service

import (
	"github.com/flipped-aurora/gin-vue-admin/server/global"
	"github.com/flipped-aurora/gin-vue-admin/server/plugin/project_manager/model"
	"github.com/flipped-aurora/gin-vue-admin/server/plugin/project_manager/model/request"
)

type ProjectService struct{}

// CreateProject 创建项目
func (s *ProjectService) CreateProject(project model.Project) error {
	return global.GVA_DB.Create(&project).Error
}

// DeleteProject 删除项目
func (s *ProjectService) DeleteProject(project model.Project) error {
	return global.GVA_DB.Delete(&project).Error
}

// UpdateProject 更新项目
func (s *ProjectService) UpdateProject(project model.Project) error {
	return global.GVA_DB.Save(&project).Error
}

// GetProject 获取项目
func (s *ProjectService) GetProject(id uint) (project model.Project, err error) {
	err = global.GVA_DB.Where("id = ?", id).First(&project).Error
	return
}

// GetProjectList 获取项目列表
func (s *ProjectService) GetProjectList(info request.ProjectSearch) (list []model.Project, total int64, err error) {
	limit := info.PageSize
	offset := info.PageSize * (info.Page - 1)

	db := global.GVA_DB.Model(&model.Project{})

	if info.Name != "" {
		db = db.Where("name LIKE ?", "%"+info.Name+"%")
	}

	err = db.Count(&total).Error
	if err != nil {
		return
	}

	err = db.Limit(limit).Offset(offset).Find(&list).Error
	return list, total, err
}
